packages/oas-normalize/src/index.ts

```diff
@@ -141,19 +140,48 @@ export default class OASNormalize {
   }
 
   /**
-   * Validate, and potentially convert to OpenAPI, a given API definition.
+   * Convert a given API definition to OpenAPI if it is not already.
+   *
+   */
+  async convert(): Promise<OpenAPI.Document> {
+    if (this.cache.convert) return Promise.resolve(this.cache.convert);
+
+    return this.load()
+      .then(async schema => {
+        // If we have a Postman collection we need to convert it to OpenAPI.
+        return utils.isPostman(schema) ? OASNormalize.convertPostmanToOpenAPI(schema) : schema;
+      })
+      .then(async schema => {
+        if (!utils.isSwagger(schema) && !utils.isOpenAPI(schema)) {
+          return Promise.reject(new Error('The supplied API definition is unsupported.'));
+        } else if (utils.isOpenAPI(schema)) {
+          return schema;
+        }
+
+        const baseVersion = parseInt(schema.swagger, 10);
+        if (baseVersion === 1) {
+          return Promise.reject(new Error('Swagger v1.2 is unsupported.'));
+        }
+
+        return converter
+          .convertObj(schema, { anchors: true })
+          .then((options: { openapi: OpenAPI.Document }) => options.openapi)
+          .catch(err => Promise.reject(err));
+      });
+  }
+
+  /**
+   * Validate a given API definition.
+   *
+   * If supplied a Postman collection it will be converted to OpenAPI first and then run through
+   * standard OpenAPI validation.
    *
    */
   async validate(
     opts: {
-      /**
-       * Automatically convert the supplied API definition to the latest version of OpenAPI.
-       */
-      convertToLatest?: boolean;
       parser?: openapiParser.Options;
-    } = { convertToLatest: false },
-  ): Promise<OpenAPI.Document> {
-    const convertToLatest = opts.convertToLatest;
+    } = {},
+  ): Promise<boolean> {
```

Comment:
this technically only ever returns `true`, correct? since it just throws an error otherwise?
```suggestion
  ): Promise<true> {
```

Reply:
TS doesn't let you do that.

![Screenshot 2024-12-20 at 9 46 08 AM](https://github.com/user-attachments/assets/59674610-6aec-4e45-bb2b-09750e8f310d)

Reply:
oh i think it's because we're returning `Promise.reject()` rather than throwing the errors — just confirmed locally that this passes:

![CleanShot 2024-12-20 at 11 52 58@2x](https://github.com/user-attachments/assets/b0673380-5770-4a7a-bc59-3c57b35d0d3c)

Reply:
hm yeah idk why i did promise.reject here. will move them over